### PR TITLE
Refactor Terraform scripts to use shared modules

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

This PR refactors our HASH Terraform config to use reusable component we've defined in another repo (yet to be a public repo). The modules we've refactored here are ones that we're using in other Terraform configs and most general.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204610208832639/1204691860848733/f) _(internal)_ (TF part of the task)

## 🔍 What does this change?

- Refactors TF modules

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] is internal and does not require a docs change 

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] does not affect the execution graph

## 🐾 Next steps

- Remove `moved` clauses from the code after the `apply` has been carried out. This will make the statements obsolete.

## 🛡 What tests cover this?

- CI/CD

## ❓ How to test this?

1.  Checkout the branch
1.  With an authenticated Vault CLI, run `terraform init` and `terraform plan --var-file prod-usea1.tfvars  --refresh=false` from `/infra/terraform/hash`
1. The plan should be _unchanged_

### CODE REVIEW

File: infra/terraform/modules/base_network/main.tf [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+  iam_role_arn    = aws_iam_role.flow_log.arn
+  log_destination = aws_cloudwatch_log_group.flow_log.arn
+  traffic_type    = "ALL"
+  vpc_id          = aws_vpc.main.id
+}
+
+resource "aws_cloudwatch_log_group" "flow_log" {
+  name = "${var.prefix}-flowlogvpc"
+}
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply infra/terraform/modules/base_network/main.tf:23]: Marking as resolved as it's a duplicate of the below

[the remark this answers is not in the document]
Comment (TimDiekmann) [reply infra/terraform/modules/base_network/main.tf:23]: Marking as resolved as it's a duplicate of the below

File: apps/hash-external-services/docker-compose.prod.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ services:
     # ports:
     #   - "5432:5432"
     volumes:
-      - hash-postgres-data:/var/lib/postgresql/data
+      - hash-postgres-data:/data/pgdata
```
Comment (thehabbos007) [inline apps/hash-external-services/docker-compose.prod.yml:33]: @indietyp drive-by here

File: infra/terraform/hash/bastion.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Bastion hosts
+
+The Bastion hosts defined through this configuration is used for SSH-ing into the private subnet of our AWS VPC.
+Bastion hosts are spun up per environment (e.g. `prod` and `dev`) and connect using current terraform outputs.
+
+It is possible to SSH into the Bastion host by running the `./ssh_bastion.sh` script.
+This script will use the currently selected terraform workspace to SSH into the appropriate host.
+
+The Bastion host can also act as a jump host to SSH tunnel a service to your local machine.
+For example to forward the RDS instance in the private subnet to your local machine, run
```
Comment (TimDiekmann) [inline infra/terraform/hash/bastion.md:10]: No strong feelings but I think we don't use newlines in other `*.md`s inside of a paragraph
```suggestion
The Bastion hosts defined through this configuration is used for SSH-ing into the private subnet of our AWS VPC. Bastion hosts are spun up per environment (e.g. `prod` and `dev`) and connect using current terraform outputs.

It is possible to SSH into the Bastion host by running the `./ssh_bastion.sh` script. This script will use the currently selected terraform workspace to SSH into the appropriate host.

The Bastion host can also act as a jump host to SSH tunnel a service to your local machine. For example to forward the RDS instance in the private subnet to your local machine, run
```

File: infra/terraform/modules/vault_aws_auth/env.sh
Comment (TimDiekmann) [inline infra/terraform/modules/vault_aws_auth/env.sh:1]: Sad that this is required but it's way less error-prone, thanks for adding it!

File: infra/terraform/modules/networking/main.tf
Comment (TimDiekmann) [inline infra/terraform/modules/networking/main.tf:1]: Do I get it correctly that this was moved to `modules/base_network`?

**Review by TimDiekmann — COMMENTED** [review]
Looks good to me. One suggestion and one question. 

I was able to test the configuration locally, however, the output was not _unchanged_ but various moves. I think this is the expected behavior?

File: infra/terraform/modules/networking/main.tf
Comment (thehabbos007) [reply infra/terraform/modules/networking/main.tf:1]: First half is moved to `modules/base_network`, second half is in `hash/networking`. The `PrivateLink` stuff is what is specific to the HASH app :)

Comment (TimDiekmann) [reply infra/terraform/modules/networking/main.tf:1]: Gotcha, thanks!

**Review by TimDiekmann — APPROVED** [review]
